samples/Azure.Analytics.Purview.Account/Generated/PurviewAccountCollections.cs

```diff
@@ -46,13 +46,14 @@ protected PurviewAccountCollections()
         /// <param name="endpoint"> The account endpoint of your Purview account. Example: https://{accountName}.purview.azure.com/account/. </param>
         /// <param name="collectionName"> The String to use. </param>
         /// <param name="apiVersion"> Api Version. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="clientDiagnostics"/>, <paramref name="pipeline"/>, <paramref name="endpoint"/>, <paramref name="collectionName"/>, or <paramref name="apiVersion"/> is null. </exception>
+        /// <exception cref="ArgumentNullException"> <paramref name="clientDiagnostics"/>, <paramref name="pipeline"/>, <paramref name="endpoint"/>, <paramref name="collectionName"/> or <paramref name="apiVersion"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="collectionName"/> is an empty string, and was expected to be non-empty. </exception>
         internal PurviewAccountCollections(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, TokenCredential tokenCredential, Uri endpoint, string collectionName, string apiVersion = "2019-11-01-preview")
         {
             Argument.AssertNotNull(clientDiagnostics, nameof(clientDiagnostics));
             Argument.AssertNotNull(pipeline, nameof(pipeline));
             Argument.AssertNotNull(endpoint, nameof(endpoint));
-            Argument.AssertNotNull(collectionName, nameof(collectionName));
+            Argument.AssertNotNullOrEmpty(collectionName, nameof(collectionName));
```

Comment:
Curious: How are we deciding when to assert not null vs. not null or empty?  I was thinking it was whether or not it was a string, but we're not using "or empty" on apiVersion.

Reply:
`apiVersion` isn't a path parameter. Strictly speaking, we probably should validate that it falls into the enum range of available versions (to avoid calling new API with the old code that can simply fail on type casting or something), but I'd prefer to have a separate story for it, cause we had no approve from archboard for it.

Reply:
Sounds good.  Would you be willing to file an issue to track this so it doesn't get lost?

Reply:
Let's first decide if we want comments for internal constructors at all.

Reply:
Fixed. Parameter validation is removed from internal constructors